server/protocols/webadmin/webadmin-data/src/main/java/org/apache/james/webadmin/routes/AliasRoutes.java

```diff
@@ -117,11 +117,17 @@ public HaltException addToAliasSources(Request request, Response response) throw
         return halt(HttpStatus.NO_CONTENT_204);
     }
 
-    private void addAlias(MappingSource source, MailAddress aliasSourceAddress) throws RecipientRewriteTableException {
+    private void addAlias(MappingSource source, MailAddress aliasSourceAddress) {
         try {
             recipientRewriteTable.addAliasMapping(source, aliasSourceAddress.asString());
         } catch (MappingAlreadyExistsException e) {
             // ignore
+        } catch (RecipientRewriteTableException e) {
```

Comment:
Not really: I would say the default here should be 500

You should create a `SameSourceAndDestinationException` being a RecipientRewriteTableException and specially catch this to handle it as a 400.